### PR TITLE
mDNS: test_app pytest migration

### PR DESCRIPTION
**Pull request by gabsuren**

### CODE REVIEW

**Review by david-cermak — CHANGES_REQUESTED** [review]
could you please rename `components/mdns/tests/test_apps/app_test.py` 
to `pytest_mdns_app.py`

File: .github/workflows/test_apps.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      - name: Build ${{ matrix.example }} with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }}
+        env: 
+          IDF_TARGET: ${{ matrix.idf_target }} 
+        shell: bash 
+        working-directory: components/mdns/tests/test_apps
+        run: | 
+          . ${IDF_PATH}/export.sh 
+          cat sdkconfig.ci >> sdkconfig.defaults  
+          idf.py set-target ${{ matrix.idf_target }}  
+          idf.py build
```
Comment (david-cermak) [inline .github/workflows/test_apps.yml:34]:
```suggestion
 $IDF_PATH/tools/ci/ci_build_apps.py <parent_dir>  --config <matrix.config> --target <target> -vv --pytest-apps
```

File: components/mdns/tests/test_apps/pytest.ini [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[pytest]
```
Comment (david-cermak) [inline components/mdns/tests/test_apps/pytest.ini:1]: Please check if we can use one `ini` file for multiple components

Comment (gabsuren) [reply components/mdns/tests/test_apps/pytest.ini]: Indeed it worked.
I remember I tried it before and it wasn't supported yet, now it is possible

File: components/mdns/tests/test_apps/app_test.py [hunk 3]
```diff
@@ -110,10 +112,10 @@ def get_dns_answer_to_service_query(mdns_service):  # type:(str) -> dpkt.dns.Msg
     arr.srvname = mdns_service
     arr.ip = socket.inet_aton('127.0.0.1')
     dns. an.append(arr)
-    console_log('Created answer to mdns query: {} '.format(dns.__repr__()))
+    print('Created answer to mdns query: {} '.format(dns.__repr__()))
     return dns.pack()
 
-
+@pytest.mark.skip
 def mdns_listener(esp_host):  # type:(str) -> None
```
[the remark this answers is not in the document]
Comment (gabsuren) [reply components/mdns/tests/test_apps/app_test.py:123]: Ok I will submit separate PR for this, thanks for the suggestion

File: pytest.ini [hunk 4]
```diff
@@ -2,11 +2,13 @@
...
 python_files = pytest_*.py
 
+# ignore PytestExperimentalApiWarning for record_xml_attribute
+# set traceback to "short" to prevent the overwhelming tracebacks
 addopts =
   -s
   --embedded-services esp,idf
-  -W ignore::_pytest.warning_types.PytestExperimentalApiWarning
   --tb short
+  --skip-check-coredump y
```
Comment (david-cermak) [inline pytest.ini:11]: I wonder why do we need to skip coredump check. Does it mean that the test app produce any? Is it possible to miss a crash with this settings?

**Review by david-cermak — APPROVED** [review]
Thanks for the update!

**Pull request merged**